File: BankAccount.java
public class BankAccount {
    public String person_name;
    public double balance;

    public BankAccount() {
        this.person_name = "";
        this.balance = 0;
    }

    public BankAccount(String n, double b) {
        person_name = n;
        balance = b;
    }

    public String getName() {
        return person_name;
    }

    public double getBalance() {
        return balance;
    }

    public void deposit(double amt) {
        balance = balance + amt;
    }

    public void withdraw(double amt) {
        balance = balance - amt;
    }

    public void transfer(BankAccount other, double amt) {
        this.withdraw(amt);
        other.deposit(amt);
    }
    @Override
    public String toString() {
        return person_name + ": " + balance;
    }
}

File: BankAccount.cpp
//
// Created by Ted Hoeller on 11/27/2018.
//

#include <iostream>
#include "BankAccount.h"
BankAccount::BankAccount() = default;

BankAccount::BankAccount(std::string n, double b) {
    this->name = n;
    this->balance = b;
}

std::string BankAccount::getName() {
    return this->name;
}

double BankAccount::getBalance() {
    return this->balance;
}

void BankAccount::deposit(double a) {
    this->balance = this->balance + a;
}

void BankAccount::withdraw(double a) {
    this->balance = this->balance - a;
}

void BankAccount::toString() {
    std::cout << name << ": " << std::to_string(balance) << std::endl;
}

void BankAccount::transfer(BankAccount *other, double b) {
    this->balance = this->balance - b;
    other->balance = other->balance + b;
}